lib/common/models/resource.rb

```diff
@@ -4,7 +4,7 @@
 require 'dry-struct'
 
 module Types
-  include Dry::Types.module
+  include Dry.Types(default: :nominal)
```

Comment:
Can we stick to Virtus and revisit this later? This will most certainly trip me up if I run into any issues.